include/Types.hpp

```diff
@@ -8,6 +8,10 @@
 #pragma once
 #include <blitz/array.h>
 
+#ifndef M_PI
+#define M_PI (3.141592653589793238)
+#endif
+
```

Comment:
Not sure this really makes sense here. I would just include <cmath> whenever numeric constants are needed, or create a new header called Constants.hpp that contains all the constants you need. +1 for checking that M_PI is defined. You can read more here: https://stackoverflow.com/questions/1727881/how-to-use-the-pi-constant-in-c?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Reply:
ok, will check it out.